File: SEP3_Client/Mediator/Information/FriendSettingPackage.cs
using SEP3_Client.Model.List.UserList;
using SEP3_Client.Model.Unit.User;

namespace SEP3_Client.Mediator.Information
{
    public class FriendSettingPackage : InformationPackage
    {
        public FriendSettingList FriendSettingList { get; set; }
        public FriendSettingPackage(FriendSettingList friendSettingList,string keyword) : base(InformationType.SETTING,keyword)
        {
            FriendSettingList = friendSettingList;
        }

        public FriendSettingPackage(FriendSetting friendSetting,string keyword) : base(InformationType.SETTING,keyword)
        {
            FriendSettingList = new FriendSettingList();
            FriendSettingList.AddNewFriendSetting(friendSetting);
        }
        
        public FriendSettingPackage()
        {
        }

        public FriendSettingList GetFriendSettingList()
        {
            return FriendSettingList;
        }
    }
}

File: SEP3_UserSystem/src/main/java/com/example/SEP3_UserSystem/mediator/information/FriendSettingPackage.java
package com.example.SEP3_UserSystem.mediator.information;

import com.example.SEP3_UserSystem.model.domain.list.userList.FriendSettingList;
import com.example.SEP3_UserSystem.model.domain.unit.user.FriendSetting;

public class FriendSettingPackage extends InformationPackage{
    private FriendSettingList FriendSettingList;

    public FriendSettingPackage(FriendSettingList friendSettingList,String keyword)
    {
        super(InformationType.SETTING,keyword);
        FriendSettingList = friendSettingList;
    }

    public FriendSettingPackage(FriendSetting friendSetting,String keyword)
    {
        super(InformationType.SETTING,keyword);
        FriendSettingList = new FriendSettingList();
        FriendSettingList.addNewFriendSetting(friendSetting);
    }

    public com.example.SEP3_UserSystem.model.domain.list.userList.FriendSettingList getFriendSettingList() {
        return FriendSettingList;
    }
}
